pkg/build/controller/config_controller.go

```diff
@@ -33,11 +37,32 @@ func IsFatal(err error) bool {
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+
+	KubeClient kclient.Interface
+	Client     osclient.Interface
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if strategy := bc.Spec.Strategy.JenkinsPipelineStrategy; strategy != nil {
+		glog.V(4).Infof("Detected Jenkins pipeline strategy in %s/%s build configuration", bc.Namespace, bc.Name)
+		if _, err := c.KubeClient.Services(bc.Namespace).Get(buildapi.JenkinsDefaultServiceName); err == nil {
```

Comment:
@jimmidyson continue will mean that when response is not not found but an 503, the Jenkins will never be created unless you modify the build config and trigger another instantiation... I think we should keep retrying in case of any error. @bparees ?